tests/components/ResultsFilter/ResultsFilter.spec.ts

```diff
@@ -42,6 +43,23 @@ describe('ResultsFilter', () => {
         expect(filter.cmp.isSelected()).toBeFalsy();
     });
 
+    it('should trigger events on toggling', done => {
+        let cpt = 0;
+        Coveo.$$(filter.env.root).on(ResultsFilterEvents.Click, (evt: Event, args: IResultsFilterEventArgs) => {
+            expect(evt.type).toBe(ResultsFilterEvents.Click);
+            if (args.checked) {
+                cpt++;
+            } else {
+                cpt++;
+            }
+            if (cpt == 2) {
+                done();
+            }
+        });
+        filter.cmp.toggle();
```

Comment:
Double toggle is normal?

Reply:
yeah alltough I could split this test in two maybe